### PR TITLE
[ETFE-3574] added consignee-trader-identification page

### PR DESCRIPTION
**Pull request by james-work-account**

### CODE REVIEW

File: conf/messages [hunk 1]
```diff
@@ -713,4 +710,13 @@ prevalidateTrader.caption = Prevalidate a trader
 prevalidateTrader.start.title = Check if a trader can receive excise goods
 prevalidateTrader.start.h1 = Check if a trader can receive excise goods
 prevalidateTrader.start.p1 = You can use this tool to check if the consignee is able to receive your goods. You will need their trader ID, such as an excise registration number (ERN) or temporary authorisation number.
-prevalidateTrader.start.button =
+
+# Prevalidate trader - consignee trader identification page
+prevalidateTrader.consigneeTraderIdentification.title = What is the consignee’s trader ID?
+prevalidateTrader.consigneeTraderIdentification.h1 = What is the consignee’s trader ID?
+prevalidateTrader.consigneeTraderIdentification.p1 = This could be an excise registration number (ERN) or a temporary authorisation reference for temporary consignees, sometimes known as a Temporary Consignment Authorisation (TCA) number.
```
Comment (LewisMitchell) [inline conf/messages:717]: I couldn't see a full stop in the copydeck, assume thats a typo in the copydeck

File: app/controllers/prevalidateTrader/PrevalidateTraderStartController.scala [hunk 2]
```diff
@@ -17,24 +17,40 @@
...
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
-import views.html.prevalidateTrader.PrevalidateTraderStartPage
+import views.html.prevalidateTrader.PrevalidateTraderStartView
 
 import javax.inject.Inject
-import scala.concurrent.ExecutionContext
+import scala.concurrent.{ExecutionContext, Future}
 
-class StartPrevalidateTraderController @Inject()(mcc: MessagesControllerComponents,
+class PrevalidateTraderStartController @Inject()(mcc: MessagesControllerComponents,
```
Comment (LewisMitchell) [inline app/controllers/prevalidateTrader/PrevalidateTraderStartController.scala:30]: Minor comment: This class could inject the navigator and do all the routing there

File: app/pages/prevalidateTrader/PrevalidateConsigneeTraderIdentificationPage.scala [hunk 3]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
```
Comment (LewisMitchell) [inline app/pages/prevalidateTrader/PrevalidateConsigneeTraderIdentificationPage.scala:2]:
```suggestion
 * Copyright 2024 HM Revenue & Customs
```

File: app/views/components/saveAndContinue.scala.html [hunk 4]
```diff
@@ -0,0 +1,36 @@
+@*
+ * Copyright 2023 HM Revenue & Customs
```
Comment (LewisMitchell) [inline app/views/components/saveAndContinue.scala.html:2]:
```suggestion
 * Copyright 2024 HM Revenue & Customs
```

File: test/navigation/PrevalidateTraderNavigatorSpec.scala [hunk 5]
```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
```
Comment (LewisMitchell) [inline test/navigation/PrevalidateTraderNavigatorSpec.scala:2]:
```suggestion
 * Copyright 2024 HM Revenue & Customs
```

File: test/controllers/prevalidateTrader/PrevalidateConsigneeTraderIdentificationControllerSpec.scala [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+            .withError(FormError("value", Seq("prevalidateTrader.consigneeTraderIdentification.error.invalidCharacters")))
+
+          status(result) mustEqual BAD_REQUEST
+          contentAsString(result) mustEqual view(
+            form = form,
+            action = controllers.prevalidateTrader.routes.PrevalidateConsigneeTraderIdentificationController.onSubmit(testErn)
+          )(userAnswersRequest(request), messages(request)).toString
+        }
+
+        "render the view when form missing" in {
```
Comment (LewisMitchell) [inline test/controllers/prevalidateTrader/PrevalidateConsigneeTraderIdentificationControllerSpec.scala:91]:
```suggestion
        "render the view when form value missing" in {
```

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: conf/messages (hunk 1 above)
Comment (james-work-account) [reply conf/messages:717]: I've asked UCD, will update the PR or copydeck with the correct version

File: app/controllers/prevalidateTrader/PrevalidateTraderStartController.scala (hunk 2 above)
Comment (james-work-account) [reply app/controllers/prevalidateTrader/PrevalidateTraderStartController.scala:30]: Matt and I discussed this, it's a trade off between keeping all navigation in one place vs extra boilerplate (creating a new Page just for this, injecting extra things into the class, etc). What do you think? I'm happy to make the change if you think it's worth it.

Comment (LewisMitchell) [reply app/controllers/prevalidateTrader/PrevalidateTraderStartController.scala:30]: Ah yeah, I didn't think about the Page object aspect. Happy to leave as is

File: conf/messages (hunk 1 above)
Comment (james-work-account) [reply conf/messages:717]: Full stop was missing in the copydeck, p1 is correct here.

**Comment by LewisMitchell** [comment]
As discussed:

- Missing `Error:` prefix on page title when in error
- Form automatically fills with valid value when an invalid value is submitted

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**